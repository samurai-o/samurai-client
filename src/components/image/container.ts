import styled from 'styled-components';

export const ImageContainer = styled.div<any>`
  width: ${({ width }) => (typeof width === 'number' ? `${width}px` : '100px')};
  border-radius: 8px;
  overflow: hidden;
`;

export const ImageDom = styled.img`
  width: inherit;
  vertical-align: middle;
`;
