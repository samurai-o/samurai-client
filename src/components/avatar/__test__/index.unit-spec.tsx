import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Avatar } from '..';
import { Image } from '../../image';

describe('avatar 组件', () => {
  it('快照测试', () => {
    const node = create(<Avatar size="middle" src="http://www.baidu.coms" />);
    expect(node).toMatchSnapshot();
  });

  it('props', () => {
    const node = shallow(<Avatar size="middle" src="http://www.baidu.coms" />);
    expect(node.find(Image).prop('src')).toEqual('http://www.baidu.coms');
    expect(node.find(Image).prop('type')).toEqual('middle');
  });
});
