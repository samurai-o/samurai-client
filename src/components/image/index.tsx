import React from 'react';
import { Size } from '../../common/interfaces';
import { ImageContainer, ImageDom } from './container';

export type ImageSize = Size;

export type ImageProps = {
  width?: number;
  type?: ImageSize;
  src?: string;
};

/**
 * 图片
 * @param props
 * @returns
 */
export function Image(props: ImageProps) {
  const { src, ..._props } = props;
  if (typeof src !== 'string') return null;
  return (
    <ImageContainer {..._props}>
      <ImageDom src={src} />
    </ImageContainer>
  );
}
