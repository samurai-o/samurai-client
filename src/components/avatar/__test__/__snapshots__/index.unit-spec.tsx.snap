// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`avatar 组件 快照测试 1`] = `
<div
  className="sc-dkPtyc jjLByl"
>
  <div
    className="sc-bdvvaa gveHwQ"
    type="middle"
    width={40}
  >
    <img
      className="sc-gsDJrp eTuZOI"
      src="http://www.baidu.coms"
    />
  </div>
</div>
`;
